IframeMediaPause/iframe_media_pausing.md

```diff
@@ -0,0 +1,189 @@
+    <title>iframe pausing</title>
+  </head>
+  <body>
+    <iframe src="https://foo.media.com" allow="media-playback-while-not-rendered 'none'"></iframe>
+  </body>
+</html>
+```
+
+In the past, the ["execution-while-not-rendered" and "execution-while-out-of-viewport"](https://wicg.github.io/page-lifecycle/#feature-policies) permission policies have been proposed as additions to the Page Lifecycle draft specification. However, these policies freeze all iframe JavaScript execution when not rendered, which is not desirable for the featured use case. Moreover, this proposal has not been adopted or standardized.
+
+## Interoperability with other Web API's
+
+Given that there exists many ways for a website to render audio in the broader Web Platform, this proposal has points of contact with many API's. To be more specific, there are two scenarios where this interaction might happen:
+- Scenario 1: When the iframe is not rendered and it attempts to play audio; and
+- Scenario 2: When the iframe is already playing audio and stops rendering. Once rendering again, to resume playback, we recommend that websites wait for a new user gesture to do so.
+
+For the first scenario, the APIs should behave as if they didn't have permission in the first place to play audio. Like when the [`autoplay` permission policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/autoplay) is set to `'none'` for an iframe. Regarding the second scenario, we propose that the API's behave as if the user had paused the media playback.
+
+The following subsections covers how this proposal could interact with Web APIs that render audio.
+
+### HTMLMediaElements
+
+HTMLMediaElement media playback is started and paused, respectively, with the [`play`](https://html.spec.whatwg.org/multipage/media.html#dom-media-play) and [`pause`](https://html.spec.whatwg.org/multipage/media.html#dom-media-pause) methods. For scenario 1, the media element shouldn't be [allowed to play](https://html.spec.whatwg.org/multipage/media.html#allowed-to-play) and `play` should return a promise rejected with `"NotAllowedError"`. In this case, the website could easily handle this like shown below.
+
+```js
+let startPlayPromise = videoElem.play();
+
+if (startPlayPromise !== undefined) {
+  startPlayPromise
+    .then(() => {
+      // Start whatever you need to do only after playback
+      // has begun.
+    })
+    .catch((error) => {
+      if (error.name === "NotAllowedError") {
+        showPlayButton(videoElem);
+      } else {
+        // Handle a load or playback error
+      }
+    });
+}
+```
+<*Snippet extracted from [MDN](https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide)*>
+
+For the scenario 2, when the iframe is not rendered anymore, the media should be [paused](https://html.spec.whatwg.org/multipage/media.html#dom-media-pause) and a [`pause`](https://html.spec.whatwg.org/multipage/media.html#event-media-pause) event should be thrown.  
```

Comment:
nit: "event should be thrown."

Exceptions are thrown.  Events are dispatched.